sdk/formrecognizer/ai-form-recognizer/review/ai-form-recognizer.api.md

```diff
@@ -0,0 +1,741 @@
+
+// @public
+export interface AnalyzeLayoutAsyncHeaders {
+    operationLocation: string;
+}
+
+// @public
+export type AnalyzeLayoutAsyncResponseModel = AnalyzeLayoutAsyncHeaders & {
+    _response: coreHttp.HttpResponse & {
+        parsedHeaders: AnalyzeLayoutAsyncHeaders;
+    };
+};
+
+// @public (undocumented)
+export type AnalyzeLayoutOperationResult = Omit<AnalyzeOperationResultModel, 'analyzeResult'> & {
+    analyzeResult?: AnalyzeLayoutResult;
+};
+
+// @public (undocumented)
+export type AnalyzeLayoutResult = Omit<AnalyzeResult, 'documentResults'>;
+
+// @public
+export type AnalyzeLayoutResultResponse = AnalyzeLayoutOperationResult & {
+    _response: coreHttp.HttpResponse & {
+        bodyAsText: string;
+        parsedBody: AnalyzeOperationResultModel;
+    };
+};
+
+// @public
+export interface AnalyzeOperationResultModel {
+    analyzeResult?: AnalyzeResultModel;
+    createdDateTime: Date;
+    lastUpdatedDateTime: Date;
+    status: OperationStatus;
+}
+
+// @public (undocumented)
+export type AnalyzeOptions = ExtractReceiptOptions | ExtractLayoutOptions | ExtractCustomFormOptions;
+
+// Warning: (ae-forgotten-export) The symbol "ICanHazStatus" needs to be exported by the entry point index.d.ts
+//
+// @public
+export type AnalyzePollerClient<T extends ICanHazStatus> = {
+    startAnalyze: (body: HttpRequestBody, contentType: SupportedContentType, analyzeOptions: AnalyzeOptions, modelId?: string) => Promise<{
+        operationLocation: string;
+    }>;
+    getAnalyzeResult: (resultId: string, options: {
+        abortSignal?: AbortSignalLike;
+    }) => Promise<T>;
+};
+
+// @public
+export type AnalyzeReceiptOperationResult = Omit<AnalyzeOperationResultModel, 'analyzeResult'> & {
+    analyzeResult?: AnalyzeReceiptResult;
+};
+
+// @public
+export type AnalyzeReceiptResult = Omit<AnalyzeResult, 'documentResults'> & {
+    receiptResults?: ReceiptResult[];
+};
+
+// @public
+export type AnalyzeReceiptResultResponse = AnalyzeReceiptOperationResult & {
+    _response: coreHttp.HttpResponse & {
+        bodyAsText: string;
+        parsedBody: AnalyzeOperationResultModel;
+    };
+};
+
+// @public (undocumented)
+export type AnalyzeResult = Omit<AnalyzeResultModel, "pageResults"> & {
+    pageResults?: PageResult[];
+};
+
+// @public
+export interface AnalyzeResultModel {
+    documentResults?: DocumentResult[];
+    errors?: ErrorInformation[];
+    pageResults?: PageResultModel[];
+    readResults: ReadResult[];
+    version: string;
+}
+
+// @public (undocumented)
+export type ArrayFieldValue = {
+    type: "array";
+    valueArray: FieldValue[];
+} & CommonFieldValue;
+
+// @public
+export class CognitiveKeyCredential implements ServiceClientCredentials {
+    constructor(apiKey: string);
+    signRequest(webResource: WebResource): Promise<WebResource>;
+    updateKey(apiKey: string): void;
+}
+
+// @public (undocumented)
+export interface CommonFieldValue {
```

Comment:
are all of these really optional?

Reply:
I think only `elements` field is optional